lms/djangoapps/discussion/rest_api/tests/test_api.py

```diff
@@ -2095,51 +2083,39 @@ def test_group_id(self, role_name, course_is_cohorted, topic_is_cohorted, data_g
             if not expected_error:
                 self.fail(f"Unexpected validation error: {ex}")
 
-    @mock.patch(
-        'lms.djangoapps.discussion.rest_api.tasks.send_thread_created_notification.apply_async'
-    )
-    def test_following(self, mock_notification_task):
+    def test_following(self):
         self.register_post_thread_response({"id": "test_id", "username": self.user.username})
         self.register_subscription_response(self.user)
         data = self.minimal_data.copy()
         data["following"] = "True"
         result = create_thread(self.request, data)
         assert result['following'] is True
-        mock_notification_task.assert_called_once()
         cs_request = httpretty.last_request()
         assert urlparse(cs_request.path).path == f"/api/v1/users/{self.user.id}/subscriptions"  # lint-amnesty, pylint: disable=no-member
         assert cs_request.method == 'POST'
         assert parsed_body(cs_request) == {'source_type': ['thread'], 'source_id': ['test_id']}
 
-    @mock.patch(
-        'lms.djangoapps.discussion.rest_api.tasks.send_thread_created_notification.apply_async'
-    )
-    def test_voted(self, mock_notification_task):
+    def test_voted(self):
         self.register_post_thread_response({"id": "test_id", "username": self.user.username})
         self.register_thread_votes_response("test_id")
         data = self.minimal_data.copy()
         data["voted"] = "True"
         with self.assert_signal_sent(api, 'thread_voted', sender=None, user=self.user, exclude_args=('post',)):
             result = create_thread(self.request, data)
         assert result['voted'] is True
-        mock_notification_task.assert_called_once()
         cs_request = httpretty.last_request()
         assert urlparse(cs_request.path).path == '/api/v1/threads/test_id/votes'  # lint-amnesty, pylint: disable=no-member
         assert cs_request.method == 'PUT'
         assert parsed_body(cs_request) == {'user_id': [str(self.user.id)], 'value': ['up']}
 
-    @mock.patch(
```

Comment:
why have all these task mocks been removed?

Reply:
Task which was being mocked has been moved to handler. These mocks have been added to signals test cases